data/service/api/v1/datasets_data_create_provenance_test.go

```diff
@@ -0,0 +1,145 @@
+package v1
```

Comment:
Package should be `v1_test` for consistency

Reply:
If I do that, then I can't call `selectXFF` or 'collectProvenanceInfo`, as they're not exported, and there's no reason for them to be.

I'm all for testing the exported APIs, but personally I have no problem with going beyond and testing non-exported functions too.

I suppose I could scaffold a bunch of junk to test selectXFF or collectProvenanceInfo via the exported API, but what does that get me? My selectXFF tests get a lot more complicated and brittle, and if selectXFF changes, I'll still have to fix all the tests.

So I'd rather keep the package v1, but if consistency is more important, reply back and I'll do the scaffolding route... Unless you have a better idea?

Reply:
This exact same issue tripped me [here](https://github.com/tidepool-org/clinic/commit/5e1381a214be06dc12963906c2dcbef3ab82f22f). I don't see any harm in exporting those functions.